File: main-task/angular/src/app/core/models/funfic.model.ts
import { Chapter } from './chapter.model';

export interface Funfic {
  id: string;
  name: string;
  author: string;
  genre: string;
  rating: number;
  scoreCount: number;
  shortDescription: string;
  chaptersCount: number;
  createdAt: string;
  tags: string[];
  chapters?: Chapter[];
}

File: main-task/webapi/Models/FunficsModels/FunficModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using webapi.Models.DbModels;

namespace webapi.Models.FunficsModels
{
    public class FunficModel
    {
        public string Id { get; set; }
        public string Author { get; set; }
        public string Name { get; set; }
        public string Genre { get; set; }
        public string ShortDescription { get; set; }
        public IEnumerable<string> Tags { get; set; }
        public byte Rating { get; set; }
        public int ScoreCount { get; set; }
        public DateTime CreatedAt { get; set; }

        public IEnumerable<PartialChapter> Chapters { get; set; }
    }
}
